### PR TITLE
fix(core): ai chat opening and append card

### PR DESCRIPTION
**Pull request by fundon**
Closes: [BS-642](https://linear.app/affine-design/issue/BS-642/continue-with-ai-：-侧边栏关闭的情况下，先唤起-侧边栏，同时插入选中的内容)

https://github.com/toeverything/AFFiNE/assets/27926/3022d808-d560-4bc8-8a04-735b76056121

### CODE REVIEW

**Comment by fundon** [comment]
* **#7324** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7324?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7322** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/7322?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fundon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx [hunk 1]
```diff
@@ -94,9 +94,22 @@ const DetailPageImpl = memo(function DetailPageImpl() {
...
+
+      if (!opened) {
+        rightSidebar.open();
+      }
+
+      // Save chat parameters:
+      // * The right sidebar is not open
+      // * Chat panel is not activated
+      if (!opened || !actived) {
+        AIProvider.sharedChatParams = params;
```
Comment (pengx17) [inline packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx]: I think you can directly modify the chat params of the chat panel component directly here without introducing sharedChatParams in AIProvider. Having a global state to be produced/consumed seems to be too frigile.

E.g., 

```typescript
// add AIProvider.requestChatPanelElement api to get the chat panel instance of the given editor
// make it promise to make sure it opens and ready to be used
type requestChatPanelElement = (host: EditorHost) => Promise<ChatPanel | null>;
AIProvider.requestChatPanelElement(host).chatParams = params;
```

Comment (fundon) [reply packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx]: There is a problem here: `ChatPanel` probably hasn't been initialized yet.

Comment (pengx17) [reply packages/frontend/core/src/pages/workspace/detail-page/detail-page.tsx]: Yeah, that's why the getter function is an async function. Could add an option to open it.

File: packages/frontend/core/src/blocksuite/presets/ai/chat-panel/chat-cards.ts [hunk 2]
```diff
@@ -443,41 +465,28 @@ export class ChatCards extends WithDisposable(LitElement) {
         if (hasImages) {
           card.images = images;
         }
-        this._updateCards(card);
+
+        if (this.cards.length === 0) {
+          this._updateCards(card);
+        } else {
+          this.cards.push(card);
```
Comment (forehalo) [inline packages/frontend/core/src/blocksuite/presets/ai/chat-panel/chat-cards.ts]: after looking into the logic of `_updateCards()`, it seems to be unnecessary for both condition.
in other word, `cards.push` and `requestUpdate` are ok for both of the branches here

**Review by forehalo — APPROVED** [review]

**Pull request merged**